src/libstd/sync/mutex.rs

```diff
@@ -293,6 +293,7 @@ impl<T: ?Sized> Mutex<T> {
 
 #[stable(feature = "rust1", since = "1.0.0")]
 impl<T: ?Sized> Drop for Mutex<T> {
+    #[unsafe_destructor_blind_to_params]
```

Comment:
yeah they are needed; otherwise the dropck-legal-cycles test won't pass.  @apasel422 [mentioned this](https://github.com/rust-lang/rust/issues/29106#issuecomment-148828533) on the other ticket, but not here.